enhancing/enhancer.js

```diff
@@ -6,15 +6,35 @@ module.exports = {
 };
 
 function succeed(item) {
+  if (item.enhancement < 20) {
+    return { ...item, enhancement: item.enhancement + 1 }
+  };
+
   return { ...item };
 }
 
 function fail(item) {
-  return { ...item };
-}
+  if (item.enhancement >= 15 && item.enhancement > 16) {
+    return {
+      ...item,
+      enhancement: item.enhancement - 1,
+      durability: Math.max(0, item.durability - 10)
+    }
+  } else if (item.enhancement >= 15) {
+    return {
+      ...item,
+      durability: Math.max(0, item.durability - 10)
+    };
+  } else {
+    return {
+      ...item,
+      durability: Math.max(0, item.durability - 5)
+    };
+  };
```

Comment:
You could alternatively break out the similarities in these checks and return at the end of the function:
```javascript
const copiedItem = { ...item };

if (item.enhancement >= 15) {
  if (item.enhancement > 16) {
    copiedItem.enhancement -= 1;
  }

  copiedItem.durability = Math.max(0, item.durability - 10);
} else {
  copiedItem.durability = Math.max(0, item.durability - 5);
}

return copiedItem;
```